src/app/components/core/configuration-manager/view-config-item/view-config-item.component.ts

```diff
@@ -34,75 +34,63 @@ export class ViewConfigItemComponent implements OnInit, OnChanges {
-            const element = currentConfigValue[0][key];
-            element.key = key;
-            configAttributes.push(element);
-          }
-        }
+        const currentConfigValues = changes.categoryConfigurationData.currentValue.value[0];
+        configAttributes = Object.keys(currentConfigValues).map(key => {
+          const element = currentConfigValues[key];
+          element.key = key;
+          return element;
+        });
+
         configAttributes = sortBy(configAttributes, function (ca) {
           return parseInt(ca.order, 10);
         });
 
         changes.categoryConfigurationData.currentValue.value = configAttributes;
         this.categoryConfiguration = changes.categoryConfigurationData.currentValue;
-        configAttributes.forEach(el => {
-          this.configItems.push({
+        this.configItems = configAttributes.map(el => {
+          return {
             key: el.key,
             value: el.value !== undefined ? el.value : el.default,
             type: el.type
-          });
+          };
         });
       }
     }
   }
 
-  public difference(obj, bs) {
-    const changedValues = differenceWith(obj, bs, (oldData: any, newData: any) => {
-      oldData.value = oldData.value === null ? 0 : oldData.value.toString();
-      newData.value = newData.key === 'integer' && newData.value === null ? 0 : newData.value.toString();
-      return oldData.key === newData.key && oldData.value === newData.value;
-    });
-
-    changedValues.forEach(element => {
-      const f = find(bs, { key: element.key });
-      if (f !== undefined) {
-        element.type = f['type'];
-      }
-    });
-    console.log('changed values', changedValues);
-    return changedValues;
-  }
-
-
   public saveConfiguration(form: NgForm) {
     this.isValidForm = true;
     if (!form.valid) {
       this.isValidForm = false;
       return false;
     }
-    const formData = [];
-    for (const key in form.value) {
-      const d = {
+
+    const formData = Object.keys(form.value).map(key => {
+      return {
         key: key,
-        value: form.value[key]
+        value: form.value[key].toString()
       };
-      formData.push(d);
-    }
-    const diff = this.difference(formData, this.configItems);
+    });
+
+    formData.map(d => {
+      return this.configItems.map(conf => {
+        if (conf.key === d.key) {
+          d['type'] = conf.type;
+        }
+        return d;
+      });
+    });
 
+    const changedConfigValues = differenceWith(formData, this.configItems, isEqual);
     // condition to check if called from add service wizard
     if (this.isWizardCall) {
-      this.onConfigChanged.emit(diff);
+      this.onConfigChanged.emit(changedConfigValues);
       return;
     }
-    diff.forEach(changedItem => {
-      this.saveConfigValue(this.categoryConfiguration.key, changedItem.key, changedItem.value, changedItem.type);
-    });
+
+    this.updateConfiguration(this.categoryConfiguration.key, changedConfigValues);
```

Comment:
there should be check, `updateConfiguration` method should not call if changedConfigValues.length == 0